lib/cachex/hook.ex

```diff
@@ -139,7 +139,9 @@ defmodule Cachex.Hook do
       end
 
       @doc false
-      def handle_info({:cachex_notify, {event, result}}, state) do
+      def handle_info({:cachex_notify, {event, result, callers}}, state) do
+        put_callers(callers)
```

Comment:
I know Elixir itself recommends this pattern, but it's really so gross in a functional language to just put state in the process state like this. Not really a request to change anything, just voicing how much I really dislike this pattern :)

Reply:
Haha, yeah, I've felt similarly!

But I got used to the fact that `Logger` does that with metadata, mocking libs do that with `$callers`, otel does that with tracing context and there are probably more cases, we don't know about where the process dictionary is used for some "spooky action at a distance". This way, the user does not have to think about passing all this stuff explicitly.

As long as it is contained to aspects like testing, logging, tracing or metrics and not used for business logic, I've learned to appreciate the pattern :)

Reply:
Yeah, it makes sense, but:

> it is contained to aspects like testing, logging, tracing or metrics

I have to include it in the production flow of my library for these cases, so it's really _not_ contained. I did consider if we could do something to make this opt-in, but that would probably cost more execution time than just blindly doing it.